AmplifyPlugins/Core/AWSPluginsCore/Model/GraphQLDocument/SingleDirectiveGraphQLDocumentBuilder.swift

```diff
@@ -0,0 +1,49 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import Amplify
+
+/// Helps construct a `SingleDirectiveGraphQLDocument`. Collects instances of the decorators and applies the changes
+/// on the document.
+public struct SingleDirectiveGraphQLDocumentBuilder {
```

Comment:
This is the first time we really start relying on a model. It's probably worth calling it out in the name of the type: `ModelBasedGraphQLlDocumentBuilder` or similar. Is `SingleDirective` super important to understanding the point of this type and how it would be used?